plugin/pkg/admission/resourcequota/controller.go

```diff
@@ -458,6 +470,43 @@ func (e *quotaEvaluator) getWork() (string, []*admissionWaiter) {
 	return ns, []*admissionWaiter{}
 }
 
+func (e *quotaEvaluator) updateCache(quota *api.ResourceQuota) {
+	e.updatedQuotasLock.Lock()
+	defer e.updatedQuotasLock.Unlock()
+
+	key := quota.Namespace + "/" + quota.Name
+	e.updatedQuotas[key] = quota
+}
+
+// checkCache compares the passes quota against the value in the look-aside cache and returns the newer
+// if the cache is out of date, it deletes the stale entry.  This only works because of etcd resourceVersions
+// being monotonically increasing integers
+func (e *quotaEvaluator) checkCache(quota *api.ResourceQuota) *api.ResourceQuota {
```

Comment:
> so this has been frowned against in the project. i am aware of no other place where we compare resource version with this assumption. is there an alternate option we could follow to do the compare?

Its done in the watch cache layer.  I'm not aware of another way to determine: "newer".